pkg/operation/botanist/controlplane.go

```diff
@@ -148,7 +149,16 @@ func (b *Botanist) HibernateControlPlane(ctx context.Context) error {
 		return err
 	}
 
-	return client.IgnoreNotFound(b.ScaleETCDToZero(ctx))
+	if err := client.IgnoreNotFound(b.ScaleETCDToZero(ctx)); err != nil {
+		return err
+	}
+
+	statefulSets := []string{
+		"etcd-main",
+		"etcd-events",
+	}
+
+	return waitUntilNoPodsExistAnymore(ctx, b.SeedClientSet.Client(), b.Shoot.SeedNamespace, nil, statefulSets)
```

Comment:
Since we are passing `[]string`, how about change the function signature to `names []string` and pass `Kind` as a parameter?

Reply:
I prefer the current approach because current approach allow waiting for `sts` and `deployment` in parallel.